pkg/virt-operator/resource/apply/crds.go

```diff
@@ -47,39 +94,24 @@ func (r *Reconciler) createOrUpdateCrd(crd *extv1beta1.CustomResourceDefinition)
 		return nil
 	}
 
+	cachedCrd = obj.(*extv1.CustomResourceDefinition)
 	if !objectMatchesVersion(&cachedCrd.ObjectMeta, version, imageRegistry, id, r.kv.GetGeneration()) {
 		// Patch if old version
-		var ops []string
-
+		for i := range crd.Spec.Versions {
+			if needsSubresourceStatusDisable(&crd.Spec.Versions[i], cachedCrd) {
+				crd.Spec.Versions[i].Subresources.Status = nil
+			}
+		}
 		// Add Labels and Annotations Patches
+		var ops []string
 		labelAnnotationPatch, err := createLabelsAndAnnotationsPatch(&crd.ObjectMeta)
 		if err != nil {
 			return err
 		}
 		ops = append(ops, labelAnnotationPatch...)
-
-		// subresource support needs to be introduced carefully after the control plane roll-over
-		// to avoid creating zombie entities which don't get processed du to ignored status updates
-		if cachedCrd.Spec.Subresources == nil || cachedCrd.Spec.Subresources.Status == nil {
-			if crd.Spec.Subresources != nil && crd.Spec.Subresources.Status != nil {
-				crd.Spec.Subresources.Status = nil
-			}
-		}
-
-		// Add Spec Patch
-		newSpec, err := json.Marshal(crd.Spec)
-		if err != nil {
+		if err := patchCRD(client, crd, ops); err != nil {
```

Comment:
nit: Let's do `return` here.

Reply:
if you did that you'd lose the logging message "crd %v is up-to-date" just below in case patching does not have any errors, I think that's pretty important.

Reply:
Fair enough :+1: